### ISSUE TITLE
License expired

### ISSUE THREAD

**Issue by Gladaed** [POE-Addon-Launcher/PAL2#51]
Rizlim Edit: PLEASE UPDATE TO THIS VERSION: https://github.com/POE-Addon-Launcher/PAL2/releases/tag/1.12.2 MANUAL UPDATE JUST OVERWRITE THE EXE!

As written 25 days ago the license is now expired (on my end)

**Comment by lagomorph42** [comment]
I'm getting the same error, java license is expired.

**Comment by FManster** [comment]
3rd

**Comment by RakivoJoy** [comment]
4th. Error started today.

Do @Rizlim buy a liecense?
Costly? Crowd fund?

**Comment by quarkslatinum** [comment]
license should be free since non commercial I would think.

same error here though

**Comment by thyll56** [comment]
I'm getting the same error, java license is expired

**Comment by keidrolis** [comment]
Same. Any update?

**Comment by FireDragonix** [comment]
Same here... :(

**Comment by seckela** [comment]
Anyone know his compile process? Might be able to 1 off compile locally from src.

**Comment by bclcburwood2** [comment]
just joining the cool crowd, also getting the expired license :(.

**Comment by Shingyboy** [comment]
yeah getting the exact same thing as everyone else.

alright guys I got something temporary whilst the owner fixes this.

open notepad and then type this 

@echo off
start "macro1.ahk" "C:\PoE Addons\Lutbot macro\macro1.ahk"
start "macro2.ahk" "C:\PoE Addons\Trade companion\macro2.ahk"
start "macro3.exe" "C:\PoE Addons\PoE Trade\macro3.exe"
exit

You can obviously edit it to your needs, where your addons are what they are called, where they are held, so if you have several and can't be bothered to open them all at once, you can even open PoE if you like.

Then save as .bat file and there you go that file will then open any program you put in there.

**Comment by POE-Addon-Launcher** [comment]
Normally it takes 3 months to expire, no idea why it did after 25 days.

**Comment by FireDragonix** [comment]
The Java licensing changed in December 2018.  This might have impacted the evaluation of versions licensing as well.... maybe?

**Comment by POE-Addon-Launcher** [comment]
Nope it's excelsior JET

**Comment by Gladaed** [comment]
Maybe you just patched PAL2 instead of also renewing the license?

**Comment by Piou1102** [comment]
Hello,
For me the message is that the trial version of java is ended.

**Comment by POE-Addon-Launcher** [comment]
Yes everyone has that issue, update is soon, just finalizing some stuff.

**Comment by POE-Addon-Launcher** [comment]
Potential Fix live, use: https://github.com/POE-Addon-Launcher/PAL2/releases/download/1.12.2/PAL2.exe

**Comment by tonpix** [comment]
This manual fix is confusing, what are we supposed to do with? Previous PAL2.exe is 30+MB, this one is 200KB, seems to be downloader/extractor, but doesn't seem to extract main .exe

**Comment by POE-Addon-Launcher** [comment]
> 
> 
> This manual fix is confusing, what are we supposed to do with? Previous PAL2.exe is 30+MB, this one is 200KB, seems to be downloader/extractor, but doesn't seem to extract main .exe

Just overwrite the old exe and it'll take care of everything, it'll download the program and then run it whenever you use the exe

If you wanna know indepth, instead of having everyone download the libraries it checks if you have java installed if you don't only then will it download the additional binaries, making the install a lot smaller for quite a lot of people, it also makes updates smaller as now to update you only need to download ~25mb whereas you had to download the full 80mb every update instead.

**Comment by tonpix** [comment]
> > This manual fix is confusing, what are we supposed to do with? Previous PAL2.exe is 30+MB, this one is 200KB, seems to be downloader/extractor, but doesn't seem to extract main .exe
> 
> Just overwrite the old exe and it'll take care of everything, it'll download the program and then run it whenever you use the exe
> 
> If you wanna know indepth, instead of having everyone download the libraries it checks if you have java installed if you don't only then will it download the additional binaries, making the install a lot smaller for quite a lot of people, it also makes updates smaller as now to update you only need to download ~25mb whereas you had to download the full 80mb every update instead.

I checked again, and got new info for you - if I double click on new .exe file, it blinks with CMD window and that's all. But when i do Ctrl+R/cmd, then drag .exe to CMD window and press enter, it launches your launcher properly. I'm not sure why there's difference with double LMB click and CMD procedure.
(of course I'm describing both scenarios after updated.exe is done with downloading and extracting)

**Comment by idmclaren** [comment]
> > > This manual fix is confusing, what are we supposed to do with? Previous PAL2.exe is 30+MB, this one is 200KB, seems to be downloader/extractor, but doesn't seem to extract main .exe
> > 
> > 
> > Just overwrite the old exe and it'll take care of everything, it'll download the program and then run it whenever you use the exe
> > If you wanna know indepth, instead of having everyone download the libraries it checks if you have java installed if you don't only then will it download the additional binaries, making the install a lot smaller for quite a lot of people, it also makes updates smaller as now to update you only need to download ~25mb whereas you had to download the full 80mb every update instead.
> 
> I checked again, and got new info for you - if I double click on new .exe file, it blinks with CMD window and that's all. But when i do Ctrl+R/cmd, then drag .exe to CMD window and press enter, it launches your launcher properly. I'm not sure why there's difference with double LMB click and CMD procedure.
> (of course I'm describing both scenarios after updated.exe is done with downloading and extracting)

Same issue. Now it just blinks open and closes and PAL does not run. I'll try running it from CMD

EDIT: That didn't work for me either, but running the JAR works to run the program. Before, running the exe would run the program. Now it does not seem to.

**Comment by quarkslatinum** [comment]
thanks for the quick fix, worked for me

**Comment by sicu** [comment]
Overwriting the exe worked fine for me as well

**Comment by 148349** [comment]
@POE-Addon-Launcher can you update the readme on how to rebuild the packages to create the zip file?

Edit: Also please read https://en.wikipedia.org/wiki/Excelsior_JET and consider moving off to graal instead. Best way to get up and running with graal is to use sdkman.

**Comment by POE-Addon-Launcher** [comment]
> @POE-Addon-Launcher can you update the readme on how to rebuild the packages to create the zip file?

If you wanna take the time to do that, then you might aswell take the time to install the JRE since that's way faster to do.

> 
> 
> 
> Edit: Also please read https://en.wikipedia.org/wiki/Excelsior_JET and consider moving off to graal instead. Best way to get up and running with graal is to use sdkman.

The issue is that because I use Kotlin all of those tools that do that never work, tried a bunch of them.

Stuff like:

- Launch4j
- https://docs.oracle.com/javase/8/docs/technotes/guides/deploy/self-contained-packaging.html
- https://github.com/libgdx/packr
- https://github.com/Jorl17/jar2app

All did not work either.

**Comment by Smooshface** [comment]
Edit: So, it works perfectly fine if I run from the same drive that Windows is installed on, but not any of my other drives. I'm assuming that has something to do with installing JRE, but I am not well enough versed in Java to really know for sure.

I can also still only get it to work if I use win+R and bring up a command prompt, then drag and drop the exe and press enter.

Essentially, it only works if I run it from command prompt, not as a stand-alone exe file.
